packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/vsphere-users/delete/vsphere-user-delete.controller.js

```diff
@@ -1,26 +1,46 @@
+import { TRACKING_TASK_TAG } from './vsphere-user-delete.constant';
+
 export default class {
   /* @ngInject */
   constructor($translate, DedicatedCloud) {
     this.$translate = $translate;
     this.DedicatedCloud = DedicatedCloud;
+    this.boundOnCloseClick = this.onCloseClick.bind(this);
+  }
+
+  $onInit() {
+    this.TRACKING_TASK_TAG = TRACKING_TASK_TAG;
+    this.task = null;
+
+    this.loaders = {
+      init: false,
+      action: false,
+      poll: false,
+    };
   }
 
   deleteUser() {
+    this.loaders.action = true;
     return this.DedicatedCloud.deleteUser(this.productId, this.user.userId)
-      .then(() => {
-        this.goBack(
-          this.$translate.instant('dedicatedCloud_USER_delete_success', {
-            t0: `<b>${this.user.name}</b>`,
-          }),
-        );
+      .then((data) => {
+        this.task = data;
+        this.loaders.action = false;
       })
-      .catch((err) => {
-        this.goBack(
+      .catch(({ data: err }) => {
+        this.loaders.action = false;
```

Comment:
**suggestion:** Instead of duplicating in `.then()` and `.catch()`. You can move the statement `this.loaders.action = false;` to `.finally()`